### PR TITLE
[WIP] Switch from sbt-microsites to docusaurus #766

### PR DESCRIPTION
**Pull request by Dan-M**
Using docusaurus to create the Microsite.
This is work in progress. 

## What's done:
- Project structure adapted and docusaurus build working locally.
- All available documentation on the microsite is available on the docusaurus generated site.

## I'm opening the PR to get feedback on the following topic:
- Change in Navigation from the old side to the new one:
  - The docusaurus site uses top navigation for the main documentation topics `Overview`, `Datatypes`, `Interop`, `Usecases` and `Advanced`
  - `Getting Started` and `Home` are not part of the top navigation anymore.
  - `Home` (the current toplevel `index.md`) is now part of the website.

Are these changes ok? Any feedback is welcome here.

## Open issues / questions:

- [x] Travis integration
- [x] Fix site nav - left: (Duplicate index entries & button navigation at the buttom of the page)
- [x] Check site nav - right: Why is it shown on some pages but not on others?
- [ ] General styling
- Check out available docusaurus features:
  - [ ] Search
  - [ ] Multi language support
  - [ ] Versioned documentation
  - [ ] Project users

### Prerequisites:
- You need `yarn` installed

### Build:
- `sbt docs/docusaurusCreateSite`

### Run locally:
- Run `yarn start` in the `website` directory and navigate to http://localhost:3000/scalaz-zio/

### CODE REVIEW

File: website/README.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
+This website was created with [Docusaurus](https://docusaurus.io/).
```
Comment (LGLO) [inline website/README.md]: Perhaps you want to delete this

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM node:8.11.4
```
Comment (LGLO) [inline Dockerfile]: Are docker files required for local docs development or for Travis? If not, I think they should be removed.

**Comment by LGLO** [comment]
From my point of view, what you asked about is OK.
I'll allow myself to comment of few other things.
1)  
 Overview link from home page doesn't work (Getting Started works correctly) 
overview-link
2)  
 top panel styling is different in Home page and in other pages
3)  
left side menu have redundant item that is working link but I feel that bigger entry above should be a link itself
left-menu
4)  
star button in footer is invalid link

**Comment by Dan-M** [comment]
Thanks for the review @LGLO 👍 
I will fix and resolve the mentioned points.

File: website/README.md (hunk 1 above)
Comment (Dan-M) [reply website/README.md]: Deleted

File: Dockerfile (hunk 2 above)
Comment (Dan-M) [reply Dockerfile]: Docker is only an option that is by the docusaurus template. It's deleted now.

File: website/i18n/en.json [hunk 3]
```diff
@@ -0,0 +1,161 @@
+{
+  "_comment": "This file is auto-generated by write-translations.js",
```
Comment (LGLO) [inline website/i18n/en.json]: If it really is, shouldn't this file be mentioned in .gitignore?

**Comment by LGLO** [comment]
@Dan-M I'll review and run locally within next 24 hours and I hope we'll get it merged soon!
Thank you very much for this effort!

**Comment by LGLO** [comment]
I'm going through pages with flow, generally it looks well. Possible improvements, IMO, I can see are:
* Getting Started has two headers, not all sections are visible in the menu on right hand side
* DataTypes (left hand side menu header) vs Datatypes (top panel) vs Data Types (left hand side menu). I think it should be "Data Types" everywhere, but I'm not native speaker
* interop (lowercase in the LHS menu), also double headers
* Usecases (LHS menu) vs Use Cases
* Broken link from Parallelism to Scheduling (NEXT button is visible but not working, usecases_queueing is missing)
* advanced (lower case in LHS menu)
* Star button in footer doesn't work - do you think we need it when we have GitHub link just above?
* do you think we need docker files present? If yes, could we hide them from top-level directory?
* website/bulid/index.html should not be commited to git

Once again, thank you for docusaurus website!

**Comment by jdegoes** [comment]
@Dan-M It would also be great to have a "home page" with the large splash image. It looks nice and I see a "splash image" on most OSS home pages.

Can you take a few quick screenshots so we can get a feel for what this looks like?

@LGLO Can you please work with Dan to get this pushed through? It'd be great if we can do that prior to the move to the new ZIO organization.

File: website/i18n/en.json (hunk 3 above)
Comment (Dan-M) [reply website/i18n/en.json]: I have added this to .gitignore.

**Comment by Dan-M** [comment]
Hey @LGLO , @jdegoes 

Sorry for the delay on this one. I had some family matters to take care of in the last 2 weeks.
I think all the points mentioned in the comment above by @LGLO  should have been addressed with the last commit.

In regards to the home page:

I'm working on getting a better presentation of the `zio` features. Looking at other pages built with `docusaurus` listed here: https://docusaurus.io/en/users I think there is 2 things that needs to be done:
- `Homesplash`: (Below the header navigation) This is usually in the secondary color and shows the project logo in a bigger fomat. Do you have any color preferences? Do you have the `zio` logo in a scalable format? That would make these changes easier.

- Feature Callouts: The current state here is just a textual list of the `zio` features (as on the old pages index.md). I will change that to the feature presentation as shown on most of the other docusaurus pages. (It would be great to have some images for the features as well in the future).

Another thing to look at would be the for open check boxes in the PR. I think at least "Search" would be something useful for the zio page because of the amount of documentation available.

**Comment by LGLO** [comment]
@jdegoes - there was reactive streams flaky test failure, I've re run this part of pipeline and Travis is green. Do you think we can merge it?
@Dan-M Are you OK with merging as it is?

**Comment by Dan-M** [comment]
@LGLO I have added one commit to better present `zio` features on the homepage.
In my opinion it can be merged and I can create issues for better styling of the home page. (I agree with @jdegoes that it is a bit lifeless as it is now). 
Additional issues to discuss would be "Search", "i18n" and "multi version documentation".

Here are some screenshots to show how it looks like now:

desktop_home

mobile_home

desktop_doc

mobile_nav

**Comment by LGLO** [comment]
@Dan-M - thank you!

**Pull request merged**